### PR TITLE
Fix more SonarCloud Related Vulnerabilities

### PR DESCRIPTION
**Pull request by SudharakaP**
This improves upon the previous PR where we fixed most of the SonarCloud related bugs.

Related to https://github.com/jhipster/jhipster-online/issues/209

### CODE REVIEW

**Comment by SudharakaP** [comment]
Some of the SonarCloud related vulnerabilities are related to not having a DTO objects for some entity classes; https://sonarcloud.io/project/issues?id=jhipster_jhipster-online&resolved=false&types=VULNERABILITY. Wondering if these are false positives? Do we need to have DTOs for this project? 🤔 

cc: @gmarziou

**Comment by SudharakaP** [comment]
I've cleaned up some code as well as added DTOs as per Sonar suggestions. Really appreciate if someone could review this. 😄 

cc: @gmarziou

File: src/main/java/io/github/jhipster/online/util/SanitizeInputs.java [hunk 1]
```diff
@@ -29,4 +29,16 @@ public static String sanitizeInput(String inputString) {
             return null;
         return inputString.replaceAll("[\n\r\t]", "_");
     }
+
+    public static boolean isAlphaNumeric(String inputString) {
+        if (inputString == null)
+            return false;
+        return inputString.matches("[a-zA-Z0-9]*");
```
Comment (gmarziou) [inline src/main/java/io/github/jhipster/online/util/SanitizeInputs.java:36]: The pattern should be compiled only once (statically)  not for each call to the method.

File: src/main/java/io/github/jhipster/online/util/SanitizeInputs.java [hunk 2]
```diff
@@ -29,4 +29,16 @@ public static String sanitizeInput(String inputString) {
...
+    }
+
+    public static boolean isLettersNumbersAndSpaces (String inputString) {
+        if (inputString == null)
+            return false;
+        return inputString.matches("[\\w\\s]*");
```
Comment (gmarziou) [inline src/main/java/io/github/jhipster/online/util/SanitizeInputs.java:42]: Static Pattern.compile() also

File: src/main/java/io/github/jhipster/online/util/SanitizeInputs.java (hunk 1 above)
Comment (SudharakaP) [reply src/main/java/io/github/jhipster/online/util/SanitizeInputs.java]: Thanks. Done 👍🏼

File: src/main/java/io/github/jhipster/online/util/SanitizeInputs.java (hunk 2 above)
Comment (SudharakaP) [reply src/main/java/io/github/jhipster/online/util/SanitizeInputs.java]: Thanks. Done 👍🏼

File: src/main/java/io/github/jhipster/online/util/SanitizeInputs.java [hunk 3]
```diff
@@ -20,13 +20,29 @@
 package io.github.jhipster.online.util;
 
 public class SanitizeInputs {
+    private static final String SANITIZE_REGEX = "[\n\r\t]";
+    private static final String ALPHANUMERIC_REGEX = "[a-zA-Z0-9]*";
```
Comment (gmarziou) [inline src/main/java/io/github/jhipster/online/util/SanitizeInputs.java:24]: The problem is still the same: the pattern is computed within String.matches(), the type of the constant should be Pattern not String.

https://www.baeldung.com/java-regex-pre-compile

Comment (SudharakaP) [reply src/main/java/io/github/jhipster/online/util/SanitizeInputs.java:24]: Ops you are right. I think I misread your second comment somehow. Thanks for pointing it out. 😄 

I've corrected this at https://github.com/jhipster/jhipster-online/pull/216

**Pull request merged**